plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt

```diff
@@ -145,6 +146,7 @@ abstract class LoginBrowser(
     }
 
     open fun loginIdC(url: String, region: AwsRegion, scopes: List<String>) {
+        val isReAuth = ToolkitConnectionManager.getInstance(project).activeConnection() != null
```

Comment:
Would this be a proper indicator of reauth, cause this is not feature-specific. So, could there be a scenario where it's authing for one feature but reauthing for another eg in the case of CC in GW and Q in IDE?

Reply:
New revision inspects scopes to distinguish between Q and non-Q